climada/engine/impact.py

```diff
@@ -48,6 +47,410 @@
 
 LOGGER = logging.getLogger(__name__)
 
+class ImpactCalc():
+    """
+    Class to compute impacts from exposures, impact function set and hazard
+    """
+
+    def __init__(self,
+                 exposures,
+                 impfset,
+                 hazard,
+                 imp_mat=None):
+        """
+        Initialize an ImpactCalc object.
+
+        The dimension of the imp_mat variable must be compatible with the
+        exposures and hazard objects.
+
+        Parameters
+        ----------
+        exposures : climada.entity.Exposures
+            exposure used to compute imp_mat
+        impf_set: climada.entity.ImpactFuncSet
+            impact functions set used to compute imp_mat
+        hazard : climada.Hazard
+            hazard used to compute imp_mat
+        imp_mat : sparse.csr_matrix, optional
+            matrix num_events x num_exp with impacts.
+            Default is an empty matrix.
+
+        Returns
+        -------
+        None.
+
+        """
+
+        imp_mat = sparse.csr_matrix(np.empty((0, 0))) if imp_mat is None else imp_mat
```

Comment:
I think setting `imp_mat` to `None` instead of an empty zero-size matrix would be a much better choice here (and in the `Impact` class). There are several instances where `imp_mat` is checked for a sensible content, but nothing beats `if imp_mat is None`, imho.
```suggestion
```

Reply:
I am not sure whether this is a good idea. Having changes in types for attributes makes it hard for methods to work with it, as you then always have to test for None. Furthermore, this would represent a change in the `Impact` class that is not fully backwards compatible.

What would be gained from having `None` instead of an empty matrix that outways the disadvantages?

Reply:
I replaced the original default argument with an empty matrix and adapted all checks to verify that `imp_mat.size > 0` in e0a6a0adb4c4f0dfb958a331ee333746a9ce9cf4